java-extension/com.microsoft.java.test.plugin/src/main/java/com/microsoft/java/test/plugin/searcher/JUnit5TestSearcher.java

```diff
@@ -110,4 +140,85 @@ public TestItem parseTestItem(IMethod method) throws JavaModelException {
         item.setParamTypes(result);
         return item;
     }
+
+    @Override
+    public TestItem parseTestItem(IMethodBinding methodBinding) throws JavaModelException {
+        final TestItem item = super.parseTestItem(methodBinding);
+
+        // deal with @DisplayName
+        for (final IAnnotationBinding annotation : methodBinding.getAnnotations()) {
+            if (annotation == null) {
+                continue;
+            }
+            if (matchesName(annotation.getAnnotationType(), DISPLAY_NAME_ANNOTATION_JUNIT5)) {
+                item.setDisplayName((String) annotation.getAllMemberValuePairs()[0].getValue());
+                break;
+            }
+        }
+
+        // parse the parameters
+        final ITypeBinding[] parameters = methodBinding.getParameterTypes();
+        if (parameters.length > 0) {
+            final List<String> result = new LinkedList<>();
+            for (final ITypeBinding parameter : parameters) {
+                result.add(parameter.getQualifiedName());
+            }
+            item.setParamTypes(result);
+        }
+
+        return item;
+    }
+
+    @Override
+    public boolean annotates(IAnnotationBinding[] annotations, String annotationName) {
+        for (final IAnnotationBinding annotation : annotations) {
+            if (annotation == null) {
+                continue;
+            }
+            if (matchesName(annotation.getAnnotationType(), annotationName)) {
+                return true;
+            }
+
+            if (JUPITER_NESTED.equals(annotationName) || JUNIT_PLATFORM_TESTABLE.equals(annotationName)) {
+                final Set<ITypeBinding> hierarchy = new HashSet<>();
+                if (matchesNameInAnnotationHierarchy(annotation, annotationName, hierarchy)) {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
+    private boolean matchesName(ITypeBinding annotationType, String annotationName) {
```

Comment:
I saw there are several places with similar logic, this can become a static utility method.

`isEquivalentAnnotationType()`